DllImportGenerator/DllImportGenerator/DllImportGenerator.cs

```diff
@@ -38,6 +40,10 @@ public void Execute(GeneratorExecutionContext context)
             var syntaxToModel = new Dictionary<SyntaxTree, SemanticModel>();
 
             var generatorDiagnostics = new GeneratorDiagnostics(context);
+            if (synRec.Methods.Any() && !IsSupportedTargetFramework(context.Compilation))
```

Comment:
Seems like we are a bit down the rabbit hole at this point. Perhaps the `synRec.Methods.Any()` should be done when we check the type is `SyntaxReceiver`?